<?xml version="1.0"?>
<launch>
    <arg name="vocab_file" default="/home/ahollenbach/catkin_ws/src/ORB_SLAM2/Vocabulary/ORBvoc.bin" />
    <arg name="camera_calib_file" default="/home/ahollenbach/catkin_ws/src/thesis/calib/orb-ar.yaml" />

    <arg name="num_clients" default="2" />
    <arg name="camera0_topic" default="/ardrone/image_raw" />
    <arg name="camera1_topic" default="/ardrone1/ardrone/image_raw" />

    <!-- little hack to run debug:=true to debug via launch -->
    <arg name="debug" default="false" />
    <arg name="debug_string" value="gdb -ex run --args" if="$(arg debug)" />
    <arg name="debug_string" value="" unless="$(arg debug)" />

    <node name="mono_server" pkg="ORB_SLAM2" type="MonoServer"
          output="screen" clear_params="true" launch-prefix="$(arg debug_string)"
          args="$(arg vocab_file) $(arg camera_calib_file) $(arg num_clients)">
        <remap from="/camera0/image_raw" to="$(arg camera0_topic)" />
        <remap from="/camera1/image_raw" to="$(arg camera1_topic)" />
    </node>
</launch>